apps/src/javalab/redux/editorRedux.ts

```diff
@@ -0,0 +1,320 @@
+/**
+ * Redux store for editor-specific Java Lab state.
+ */
+
+// TODO: Can we fix our imports and no longer need to ignore this rule?
+/* eslint-disable @typescript-eslint/no-var-requires */
+
+const {
+  fileMetadataForEditor,
+  updateAllSourceFileOrders
+} = require('@cdo/apps/javalab/JavalabFileHelper');
+import _ from 'lodash';
+import {createSelector, createSlice, PayloadAction} from '@reduxjs/toolkit';
+import {JavalabEditorDialog} from '../types';
+
+interface Sources {
```

Comment:
nit: maybe `SourcesMap` or `EditorFilesMap`, or something more specific to Javalab? Thinking about how we'll soon have a `Source` type in lab lab as well

Reply:
oh good call, I'll make this name more specific